### PR TITLE
Fix Incompatibilities with PHP 7 & 8

### PR DESCRIPTION
**Pull request by tn3rb**
This Add-on throws fatal errors in PHP 8 and warnings in PHP 7

This PR:

- fixes incompatibilities with current PHP versions
- performs  tons of code clean up
- adds php 7 typing
- auto formats everything

### CODE REVIEW

File: src/application/services/import/config/ImportConfigBase.php [hunk 1]
```diff
@@ -62,43 +59,50 @@ protected function checkModelConfigsInitialized()
...
-     * @since 1.0.0.p
+     *
      * @return mixed
+     * @since 1.0.0.p
      */
     public function toJsonSerializableData()
     {
-        $simple_obj = new stdClass();
+        $simple_obj                     = new stdClass();
         $simple_obj->json_model_configs = [];
```
Comment (hoseinrafiei) [inline src/application/services/import/config/ImportConfigBase.php:82]: 👍

File: src/domain/services/commands/ImportAttendeeCommand.php [hunk 2]
```diff
@@ -14,25 +14,23 @@
...
     private $registration;
 
-    public function __construct(
-        EE_Registration $reg,
-        array $input_data,
-        ImportAttendeeConfig $config
-    ) {
+
+    public function __construct(EE_Registration $reg, array $input_data, ImportAttendeeConfig $config)
+    {
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply src/domain/services/commands/ImportAttendeeCommand.php:24]: i have wrapping set up at 120 characters. If a line exceeds that length then it should wrap.
my apologies for never mentioning that before

File: src/ui/admin/importer/help_tabs/importer_import_map.help_tab.php [hunk 3]
```diff
@@ -42,5 +44,5 @@
             // @codingStandardsIgnoreEnd
             'event_espresso'
         );
-    ?>
+        ?>
 </p>
```
[the remark this answers is not in the document]
Comment (tn3rb) [reply src/ui/admin/importer/help_tabs/importer_import_map.help_tab.php:48]: thnx, will fix

File: src/domain/services/commands/ImportAttendeeCommand.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (hoseinrafiei) [reply src/domain/services/commands/ImportAttendeeCommand.php:24]: Never mind. Do you use PHPStorm for reformatting automatically? If no which tool do you use?

[the remark this answers is not in the document]
Comment (tn3rb) [reply src/domain/services/commands/ImportAttendeeCommand.php:24]: yes I use phpStorm
years ago all of the EE devs had their settings synced via an online repo
but that hasn't been updated in ages
i can look into exporting my settings if you want to use them

[the remark this answers is not in the document]
Comment (hoseinrafiei) [reply src/domain/services/commands/ImportAttendeeCommand.php:24]: That would be great. Thanks.

**Pull request merged**